lib/payload/scanner/SSL_TLS/engine.py

```diff
@@ -0,0 +1,674 @@
+        cipher['\xC0\xAF'] = 'TLS_ECDHE_ECDSA_WITH_AES_256_CCM_8'
+        cipher['\xCC\x13'] = 'TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256'
+        cipher['\xCC\x14'] = 'TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256'
+        cipher['\xCC\x15'] = 'TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256'
+        # http://www.mozilla.org/projects/security/pki/nss/ssl/fips-ssl-ciphersuites.html
+        cipher['\xFE\xFE'] = 'SSL_RSA_FIPS_WITH_DES_CBC_SHA'
+        cipher['\xFE\xFF'] = 'SSL_RSA_FIPS_WITH_3DES_EDE_CBC_SHA'
+        cipher['\xFF\xE0'] = 'SSL_RSA_FIPS_WITH_3DES_EDE_CBC_SHA'
+        cipher['\xFF\xE1'] = 'SSL_RSA_FIPS_WITH_DES_CBC_SHA'
+        # Special one to indicate client's support for downgrade prevention
+        cipher['\x56\x00'] = 'TLS_FALLBACK_SCSV'
+        return cipher
+
+    def __helloRand(self):
+        rand = struct.pack("!L", int(time.time()))
+        rand += os.urandom(28)
+        return rand
+
+    def ssl2Hello(self, cipher = 0):
+        cipher_spec = ""
+        if cipher == 0:
+            cipher_list = self.ssl2Ciphers()
+            for c in cipher_list.keys():
+                cipher_spec += c
+        else:
+            for c in cipher.keys():
+                cipher_spec += c
+        challenge = "\x6a\x61\x6e\x73\x65\x6e\x6f\x66\x6c\x6f\x72\x6b\x65\x65\x72\x73"
+        p_len = len(cipher_spec) + len(challenge) + 9
+
+        mask = bin(0x8000) # Most significant bit should be set to indicate no padding
+        bin_plen = bin(p_len)
+
+        packet = ""
+        packet = struct.pack("!H", int(mask,2) + int(bin_plen,2))  # Length of record
+        packet += "\x01" 
+        packet += "\x00\x02"
+        packet += struct.pack("!H", len(cipher_spec))
+        packet += "\x00\x00"
+        packet += "\x00\x10" 
+        packet += cipher_spec
+        packet += challenge
+        return packet
+
+    def tlsHello(self, protocol, cipher = 0):
+        record = "\x16" # Message Type 22
+        record += self.protocols[protocol]
+        cipher_suites = ""
+        if cipher == 0:
+            cipher_list = self.tlsCiphers()
+            for c in cipher_list.keys():
+                cipher_suites += c
+        else:
+            for c in cipher.keys():
+                cipher_suites += c
+        if (len(cipher_suites) % 2) == 0:
+            compression = "\x02\x01\x00"
+            hp_len = len(cipher_suites) + len(compression) + 41
+            record += struct.pack("!H", hp_len)
+            handshake = "\x01" # Client Hello Message
+            h_len = struct.pack("!L", hp_len-4)
+            handshake += h_len[1:]
+            handshake += self.protocols[protocol]
+            handshake += self.__helloRand()
+            handshake += "\x00"
+            handshake += struct.pack("!H", len(cipher_suites))
+            handshake += cipher_suites
+            handshake += compression
+            return record+handshake
+        else:
+            #Error in cipher length!
```

Comment:
According to PEP8, there should be a space between `#` and the comment. There are many comments with the same issue.